File: fix_conflicts.py
import os

def resolve_conflict_accept_incoming(file_path):
    print(f"Processing {file_path}")
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            lines = f.readlines()
        
        new_lines = []
        state = 'NORMAL' # NORMAL, HEAD, INCOMING
        
        for line in lines:
            if line.startswith('<<<<<<< HEAD'):
                state = 'HEAD'
                continue
            elif line.startswith('======='):
                if state == 'HEAD':
                    state = 'INCOMING'
                else:
                    # Depending on diff3 or other formats, ======= might appear. 
                    # Assuming standard format: HEAD then INCOMING.
                    state = 'INCOMING' 
                continue
            elif line.startswith('>>>>>>>'):
                state = 'NORMAL'
                continue
            
            if state == 'NORMAL':
                new_lines.append(line)
            elif state == 'INCOMING':
                new_lines.append(line)
            elif state == 'HEAD':
                # Skip HEAD content
                pass
        
        with open(file_path, 'w', encoding='utf-8') as f:
            f.writelines(new_lines)
        print(f"Resolved {file_path}")
        
    except Exception as e:
        print(f"Error processing {file_path}: {e}")

base_dir = r"c:\Users\samee\OneDrive\Desktop\AESTR-ALPHA\AestrAlpha\src\app"
files_to_fix = [
    os.path.join(base_dir, "apple-ecosystem-residency", "page.tsx"),
    os.path.join(base_dir, "modern-enterprise-backend-residency", "page.tsx")
]

for file_path in files_to_fix:
    resolve_conflict_accept_incoming(file_path)

File: fix_conflicts.js
const fs = require('fs');
const path = require('path');

function resolveConflict(filePath) {
    console.log(`Processing ${filePath}`);
    try {
        const content = fs.readFileSync(filePath, 'utf-8');
        const lines = content.split('\n');

        let newLines = [];
        let state = 'NORMAL'; // NORMAL, HEAD, INCOMING

        for (const line of lines) {
            if (line.trim().startsWith('<<<<<<< HEAD')) {
                state = 'HEAD';
                continue;
            } else if (line.trim().startsWith('=======')) {
                if (state === 'HEAD') {
                    state = 'INCOMING';
                } else {
                    // unexpected, but treat as incoming start if not in head
                    state = 'INCOMING';
                }
                continue;
            } else if (line.trim().startsWith('>>>>>>>')) {
                state = 'NORMAL';
                continue;
            }

            if (state === 'NORMAL') {
                newLines.push(line);
            } else if (state === 'INCOMING') {
                newLines.push(line);
            } else if (state === 'HEAD') {
                // Skip HEAD content
            }
        }

        fs.writeFileSync(filePath, newLines.join('\n'), 'utf-8');
        console.log(`Resolved ${filePath}`);

    } catch (e) {
        console.error(`Error processing ${filePath}: ${e.message}`);
    }
}

const baseDir = String.raw`c:\Users\samee\OneDrive\Desktop\AESTR-ALPHA\AestrAlpha\src\app`;
const filesToFix = [
    path.join(baseDir, "apple-ecosystem-residency", "page.tsx"),
    path.join(baseDir, "modern-enterprise-backend-residency", "page.tsx")
];

filesToFix.forEach(resolveConflict);
